disco/preprocess/select_timepoints2.py

```diff
@@ -166,166 +310,247 @@ def get_profile_data():
-                             if 'critical_time_idx' in dic.keys()]
+
+    critical_time_indices = [
+        t
+        for bus, dic in ag_series.items()
+        for t in dic["critical_time_idx"]
+        if "critical_time_idx" in dic
+    ]
     critical_time_indices += head_critical_time_indices
-    critical_time_indices = list(set(critical_time_indices))
-    critical_time_indices.sort()
-    destination_model_dir = os.path.join(destination_dir,'new_model')
-    os.makedirs(destination_model_dir, exist_ok=True)
+    critical_time_indices = sorted(set(critical_time_indices))
+    compression_rate = 0
     if recreate_profiles:
-        destination_profile_dir = os.path.join(destination_dir,'new_profiles')
-        os.makedirs(destination_profile_dir, exist_ok=True)
-        
-    for profile, val in profile_data.items():
-        if profile in used_profiles:
-            base_len = len(val['time_series'])
-            compression_rate = len(critical_time_indices)/base_len
-            if recreate_profiles:
-                data = val['time_series'][critical_time_indices]
-                new_profile_path = os.path.join(destination_profile_dir, f'{profile}.csv')
-                pd.DataFrame(data).to_csv(new_profile_path, index=False, header=False)
-            
-    reset_profile_data(used_profiles, critical_time_indices)
-    save_circuit(destination_model_dir)
-            
+        destination_profile_dir = destination_dir / "new_profiles"
+        destination_profile_dir.mkdir()
+
+        for profile, val in profile_data.items():
+            if profile in used_profiles:
+                base_len = len(val["time_series"])
+                compression_rate = len(critical_time_indices) / base_len
+                if recreate_profiles:
+                    data = val["time_series"][critical_time_indices]
+                    new_profile_path = destination_profile_dir / f"{profile}.csv"
+                    pd.DataFrame(data).to_csv(new_profile_path, index=False, header=False)
+
+    if create_new_circuit:
+        reset_profile_data(used_profiles, critical_time_indices)
+        before_path = destination_dir / "power_flow_results_before"
+        after_path = destination_dir / "power_flow_results_after"
+        destination_model_dir = destination_dir / "new_model"
+        destination_model_dir.mkdir()
+        save_circuit(destination_model_dir)
+        master_file = destination_model_dir / "Master.dss"
+        with open(master_file, "a") as f_out:
+            f_out.write("Calcvoltagebases\n")
+            f_out.write("Solve\n")
+
     return ag_series, head_critical_time_indices, critical_time_indices, compression_rate
-    
 
-def main(path_to_master, 
-         category_class_dict, 
-         critical_conditions=['max_demand', 'min_demand', 'max_generation', 'max_net_generation'],
-         recreate_profiles=False,
-         destination_dir=None):
+
+def get_metadata(ag_series, head_critical_time_indices):
```

Comment:
I added a function to collect metadata. My thought was to get all buses and conditions associated with a specific timepoint. @KwamiS , could you review, I want to confirm that I have interpreted the outputs correctly.

also, any comments on the format?
@daniel-thom , @KwamiS